File: WEEK1/DAY6/Prog5/P5.java
import java.util.*;
public class P5 {
    public static int factorial(int n) {
        if(n==0 || n==1){
            return 1;
        }
        return n*factorial(n-1);
    }
    public static void main(String[] args) {
        int N;
        Scanner sc=new Scanner(System.in);
        System.out.println("Check if N is strong number");
        System.out.print("Enter N: ");
        N=sc.nextInt();
        int n=N;
        int dig;
        int ans=0;
        while(n>0){
            dig=n%10;
            ans+=factorial(dig);
            n/=10;
        }
        if(ans==N){
            System.out.println("Strong number");
        }
        else{
            System.out.println("Not strong number");
        }
        sc.close();
    }
}

File: WEEK1/DAY6/Prog5/P5.py
def factorial(n):
    if n==1 or n==0:
        return 1
    return n*factorial(n-1)

print("Check if N is strong number")
N=int(input("Enter N: "))
n=N
ans=0
while n>0:
    dig=n%10
    ans+=factorial(dig)
    n//=10
if(ans==N):
    print("Strong number")
else:
    print("Not strong number")
